Artemis/Artemis/DeviceProviders/Arduino/ArduinoStrip.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+using System.Threading;
+using System.Management;//added Management for detecting Arduino
+using Artemis.DAL;
+using Artemis.Properties;
+using Artemis.Settings;
+using Artemis.Utilities;
+using System.Collections.Generic;
+using System.Text;
+using System.IO.Ports;
+using Ninject.Extensions.Logging;
+using System.IO;
+
+namespace Artemis.DeviceProviders.Arduino
+{
+	public class ArduinoStrip : DeviceProvider
+	{
+		private SerialPort _port;
+		const int Width = 288;
+		const int Height = 1;
+		private byte[] RGBColors = Enumerable.Repeat((byte)0x20, Width * 3).ToArray();
+
+		public ArduinoStrip(ILogger logger)
+		{
+			Logger = logger;
+			Type = DeviceType.Generic;
+		}
+
+		public ILogger Logger { get; set; }
+
+		public override void UpdateDevice(Bitmap bitmap)
+		{
+			if (bitmap == null)
+				return;
+
+			// Resize the bitmap
+			using (var b = ImageUtilities.ResizeImage(bitmap, Width, Height))
+			{
+				for (var x = 0; x < Width; x++)
+				{
+					var c = b.GetPixel(x, 0);
+					RGBColors[(x * 3) + 0] = c.R;
+					RGBColors[(x * 3) + 1] = c.G;
+					RGBColors[(x * 3) + 2] = c.B;
+				}
+			}
+			_port.Write(RGBColors, 0, Width * 3);
+		}
+
+		private string AutodetectArduinoPort() //from Brandon https://stackoverflow.com/questions/3293889/how-to-auto-detect-arduino-com-port
+		{
+			ManagementScope connectionScope = new ManagementScope();
+			SelectQuery serialQuery = new SelectQuery("SELECT * FROM Win32_SerialPort");
+			ManagementObjectSearcher searcher = new ManagementObjectSearcher(connectionScope, serialQuery);
+
+			foreach (ManagementObject item in searcher.Get())
+			{
+				string desc = item["Description"].ToString();
+				string deviceId = item["DeviceID"].ToString();
+
+				if (desc.Contains("Arduino"))//my due comes up as Arduino Due
+				{
+					return deviceId;
+				}
+				if (desc.Contains("USB Serial Device"))//my pro micro come up as this
+				{
+					return deviceId;
+				}
+			}
+			return null;
+		}
+
+		public override bool TryEnable()
```

Comment:
@SpoinkyNL correct me if I'm wrong, but we want to set CanUse to true _only_ if the serial port correctly selects the arduino. If lines 84-86 fail we're left with a device Artemis _thinks_ will work but in fact does not.

Reply:
Yeah that's true now it'll just always assume the device is available. Admittedly the device providers framework is a bit odd (it's got stretch marks all over it) but make sure CanUse is only set to true if you're sure Artemis can talk to the Arduino

Reply:
I will try changing it but is there a way to change the CanUse in UpdateDevice for when arduino get disconnected?

Reply:
Yeah, you can just write to it whenever you want from anywhere within the DeviceProvider class, Artemis checks the CanUse value each frame [here](https://github.com/SpoinkyNL/Artemis/blob/master/Artemis/Artemis/Managers/LoopManager.cs#L163)